src/Features/Core/Portable/Diagnostics/DiagnosticAnalyzerInfoCache.cs

```diff
@@ -85,6 +85,11 @@ internal sealed partial class DiagnosticAnalyzerInfoCache
         /// </remarks>
         private readonly ConditionalWeakTable<DiagnosticAnalyzer, DiagnosticDescriptorsInfo> _descriptorsInfo;
 
+        /// <summary>
+        /// Information about host analyzers that can be skipped for the given project ID.
+        /// </summary>
+        private readonly ConditionalWeakTable<ProjectId, ISkippedAnalyzersInfo> _skippedAnalyzersInfo;
```

Comment:
❔ Any reason to not use the same `IReadOnlyList<AnalyzerReference>` key that `CodeFixService` uses?

Reply:
We only have the project ID in the project removal callback when we clear this cache entry.

Reply:
Wouldn't it clear itself if we use CWT?